src/main/java/world/bentobox/bentobox/managers/AddonsManager.java

```diff
@@ -154,10 +162,42 @@ private void loadAddon(@NonNull File f) {
      * Enables all the addons
      */
     public void enableAddons() {
-        if (!getLoadedAddons().isEmpty()) {
-            plugin.log("Enabling addons...");
-            getLoadedAddons().forEach(this::enableAddon);
-            plugin.log("Addons successfully enabled.");
+        if (getLoadedAddons().isEmpty()) return;
+        plugin.log("Enabling addons...");
+        getLoadedAddons().forEach(this::enableAddon);
+        // Set perms for enabled addons
+        this.getEnabledAddons().forEach(this::setPerms);
+        plugin.log("Addons successfully enabled.");
+    }
+
+    void setPerms(Addon addon) {
+        ConfigurationSection perms = addon.getDescription().getPermissions();
+        perms.getKeys(true).parallelStream().filter(perm -> perms.contains(perm + DEFAULT) && perms.contains(perm + ".description"))
+        .forEach(perm -> {
+            try {
+                registerPermission(perms, perm);
+            } catch (InvalidAddonDescriptionException e) {
+                plugin.logError("Addon " + addon.getDescription().getName() + ": " + e.getMessage());
+            }
+        });
+    }
+
+    void registerPermission(ConfigurationSection perms, String perm) throws InvalidAddonDescriptionException {
+        if (perms.getString(perm + DEFAULT) == null) {
+            throw new InvalidAddonDescriptionException("Permission default is invalid : " + perms.getName());
+        }
+        PermissionDefault pd = PermissionDefault.getByName(perms.getString(perm + DEFAULT));
+        if (pd == null) {
+            throw new InvalidAddonDescriptionException("Permission default is invalid : " + perms.getName());
+        }
+        String desc = perms.getString(perm + ".description");
+        // Replace placeholders for Game Mode Addon names
+        if (perm.contains(GAMEMODE)) {
+            this.getGameModeAddons().stream().map(g -> g.getDescription().getName().toLowerCase())
```

Comment:
Shouldn't it be WorldSettings#getPermissionPrefix() method?

Reply:
Yes, thanks!